src/statmodels.jl

```diff
@@ -420,6 +420,23 @@ mutable struct CoefTable
     end
 end
 
+Base.length(ct::CoefTable) = length(ct.cols[1])
+function Base.eltype(ct::CoefTable)
+    nmtype = isempty(ct.rownms) ? String : eltype(ct.rownms)
+    NamedTuple{(Symbol("Name"), Symbol.(ct.colnms)...),
```

Comment:
The other option is we could return a custom "view" struct here instead of a NamedTuple. That could make it a bit more efficient and is a common pattern for other "table" types. For example, it's common to have something like:
```julia
struct CoefRow
    table::CoefTable
    rownumber::Int
end
```
Then on top of that you just need to implement `propertynames` and `getproperty` that correspond to the "row values" to be returned. The property names could be computed a single time in the initial `Base.iterate(ct::CoefTable)` method and passed as an additional piece of iterator state. It's obviously a bit more code/work, so maybe not worth it if these tables are always going to be small and simple anyway.

Reply:
Interesting, thanks. Though these tables are always super small so as you say I'm not sure it's worth the additional complexity. If performance really mattered we should implement a column-wise approach anyway.